samples/storage/src/main/java/com/example/platform/storage/mediastore/MediaStoreQuery.kt

```diff
@@ -0,0 +1,169 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.example.platform.storage.mediastore
+
+import android.Manifest.permission.READ_EXTERNAL_STORAGE
+import android.Manifest.permission.READ_MEDIA_IMAGES
+import android.annotation.SuppressLint
+import android.content.ContentResolver
+import android.content.ContentUris
+import android.os.Build
+import android.provider.MediaStore
+import android.provider.MediaStore.Images
+import android.text.format.Formatter
+import androidx.annotation.RequiresPermission
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Divider
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.example.platform.base.PermissionBox
+import com.google.android.catalog.framework.annotations.Sample
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+@Sample(
+    name = "MediaStore - Query",
+    description = "Query files indexed by MediaStore",
+    documentation = "https://developer.android.com/training/data-storage/shared/media#media_store",
+)
+@SuppressLint("MissingPermission")
+@Composable
+fun MediaStoreQueryScreen() {
+    val permission = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        READ_MEDIA_IMAGES
+    } else {
+        READ_EXTERNAL_STORAGE
+    }
+
+    PermissionBox(
+        permission = permission,
+        onGranted = { MediaStoreQueryContent() },
+        description = "Permission needed to access the device's photo library",
+    )
+}
+
+@RequiresPermission(anyOf = [READ_MEDIA_IMAGES, READ_EXTERNAL_STORAGE])
+@Composable
+fun MediaStoreQueryContent() {
+    val context = LocalContext.current
+    var files by remember { mutableStateOf(emptyList<FileEntry>()) }
```

Comment:
I think the recommendation for this is to use [produceState](https://developer.android.com/jetpack/compose/side-effects#producestate). You can make files immutable then